fix(vue): Cast name parameter to string

Bobakanoosh opened the pull request:
Before submitting a pull request, please take a look at our
[Contributing](https://github.com/getsentry/sentry-javascript/blob/master/CONTRIBUTING.md) guidelines and verify:

- [x] If you've added code that should be tested, please add tests.
- [x] Ensure your code lints and the test suite passes (`yarn lint`) & (`yarn test`).

Solves https://github.com/getsentry/sentry-javascript/issues/4482

AbhiPrasad reviewed (changes requested):
Instead of changing the types for all the transactions, let's just fix this for Vue for now!

Can we change https://github.com/getsentry/sentry-javascript/blob/f89a842c007e894b1c9f25db604e00d7753bd4e2/packages/vue/src/router.ts#L56 and https://github.com/getsentry/sentry-javascript/blob/f89a842c007e894b1c9f25db604e00d7753bd4e2/packages/vue/src/router.ts#L66 to always pass strings?

Bobakanoosh commented:
That's a much more elegant solution lol. Updated to reflect your changes, I also found one more usage in
`/packages/ember/addon/instance-initializers/sentry-performance.ts`

and fixed it there

AbhiPrasad commented on packages/ember/addon/instance-initializers/sentry-performance.ts:
```diff
@@ -37,7 +37,7 @@ function getBackburner() {
 function getTransitionInformation(transition: any, router: any) {
   const fromRoute = transition?.from?.name;
-  const toRoute = transition && transition.to ? transition.to.name : router.currentRouteName;
+  const toRoute = transition && transition.to ? transition.to.name.toString() : router.currentRouteName;
```
Let’s avoid changing ember for now since the router should always return a string? We can come back to this later on.

Bobakanoosh replied:
Fixed

AbhiPrasad reviewed (approved):
Awesome, thank you for your contribution!

This won't be part of the next release as it is in progress (https://github.com/getsentry/publish/issues/779), but it'll be part of the release after that (`6.17.5` or `6.18.0`)

Bobakanoosh commented:
Any idea how long until that release? I really want to use this feature of Sentry but I can't until then

AbhiPrasad commented:
We can try to cut another patch release soon then!

JoseOrtiz commented:
There was an issue here, as `name` is not always used, `to.name` is undefined, so `to.name.toString()` will fail when used in Vue

AbhiPrasad commented:
@JoseOrtiz Good catch, will open a PR to fix!

AbhiPrasad commented:
Opened PR to address this: https://github.com/getsentry/sentry-javascript/pull/4530, will cut a patch release right afterwards!

snoozbuster commented:
hey @AbhiPrasad - I think this PR has another issue. `to.name` is not necessarily defined. I think in both of these cases, `to.name?.toString()` is probably the correct line of code (or, `(to.name || to.path).toString()`). in our case we have a 404 route that isn't named and it is causing a crash similar to the symbolic route issue. I should be able to mitigate this by adding a name to the route, but thought it was worth pointing out.

AbhiPrasad commented:
This is still occurring with the fix in https://github.com/getsentry/sentry-javascript/pull/4530 @snoozbuster? That PR should mean we always check to `to.name`'s existence.

PRs are welcome if we need to introduce additional guards.
